src/spatial.jl

```diff
@@ -462,6 +464,110 @@ function psiangle(chain::Chain, res_id::Union{Integer, AbstractString})
     return psiangle(chain[resids(chain)[i]], chain[resids(chain)[i + 1]])
 end
 
+# source: http://www.mlb.co.jp/linux/science/garlic/doc/commands/dihedrals.html
+const chitables = [
+    Dict{String,NTuple{4,String}}(
+        "ARG" => ("N", "CA", "CB", "CG"),
+        "ASN" => ("N", "CA", "CB", "CG"),
+        "ASP" => ("N", "CA", "CB", "CG"),
+        "CYS" => ("N", "CA", "CB", "SG"),
+        "GLN" => ("N", "CA", "CB", "CG"),
+        "GLU" => ("N", "CA", "CB", "CG"),
+        "HIS" => ("N", "CA", "CB", "CG"),
+        "ILE" => ("N", "CA", "CB", "CG1"),
+        "LEU" => ("N", "CA", "CB", "CG"),
+        "LYS" => ("N", "CA", "CB", "CG"),
+        "MET" => ("N", "CA", "CB", "CG"),
+        "PHE" => ("N", "CA", "CB", "CG"),
+        "PRO" => ("N", "CA", "CB", "CG"),
+        "SER" => ("N", "CA", "CB", "OG"),
+        "THR" => ("N", "CA", "CB", "OG1"),
+        "TRP" => ("N", "CA", "CB", "CG"),
+        "TYR" => ("N", "CA", "CB", "CG"),
+        "VAL" => ("N", "CA", "CB", "CG1"),
+    ),
+    Dict{String,NTuple{4,String}}(
+        "ARG" => ("CA", "CB", "CG", "CD"),
+        "ASN" => ("CA", "CB", "CG", "OD1"),
+        "ASP" => ("CA", "CB", "CG", "OD1"),
+        "GLN" => ("CA", "CB", "CG", "CD"),
+        "GLU" => ("CA", "CB", "CG", "CD"),
+        "HIS" => ("CA", "CB", "CG", "ND1"),
+        "ILE" => ("CA", "CB", "CG1", "CD1"),
+        "LEU" => ("CA", "CB", "CG", "CD1"),
+        "LYS" => ("CA", "CB", "CG", "CD"),
+        "MET" => ("CA", "CB", "CG", "SD"),
+        "PHE" => ("CA", "CB", "CG", "CD1"),
+        "PRO" => ("CA", "CB", "CG", "CD"),
+        "TRP" => ("CA", "CB", "CG", "CD1"),
+        "TYR" => ("CA", "CB", "CG", "CD1"),
+    ),
+    Dict{String,NTuple{4,String}}(
+        "ARG" => ("CB", "CG", "CD", "NE"),
+        "GLN" => ("CB", "CG", "CD", "OE1"),
+        "GLU" => ("CB", "CG", "CD", "OE1"),
+        "LYS" => ("CB", "CG", "CD", "CE"),
+        "MET" => ("CB", "CG", "SD", "CE"),
+    ),
+    Dict{String,NTuple{4,String}}(
+        "ARG" => ("CG", "CD", "NE", "CZ"),
+        "LYS" => ("CG", "CD", "CE", "NZ"),
+    ),
+    Dict{String,NTuple{4,String}}(
+        "ARG" => ("CD", "NE", "CZ", "NH1"),
+    ),
+]
+
+# source: Jumper et al 2021, Supp. Table 2
+const chisymmetries = [
+    ("ASP", 2),
+    ("GLU", 3),
+    ("PHE", 2),
+    ("TYR", 2),
+]
+
+function chiangle(a1, a2, a3, a4, sym::Bool)
+    χ = dihedralangle(a1, a2, a3, a4)
+    if sym && χ < 0
+        χ += π
+    end
+    return χ
+end
+
+"""
+    chiangle(res, i)
+
+Calculate the χᵢ angle in radians for a standard `AbstractResidue` with standard atom names.
+The angle is in the range -π to π.
+"""
+function chiangle(res::AbstractResidue, i::Integer)
+    1 <= i <= 5 || throw(ArgumentError("Chi angle must be between 1 and 5"))
+    ct = chitables[i]
+    rname = resname(res)
+    t = get(ct, rname, nothing)
+    t === nothing && throw(ArgumentError("Chi angle $i does not exist for residue $rname"))
```

Comment:
Maybe check the residue name is a standard amino acid and throw a different error if it isn't, e.g. `"Chi angles are only defined for standard amino acids, residue name is $rname"`.